File: smart_planner_web_frontend/src/test-api.js
// src/test-api.js
export async function testBackendConnection() {
  try {
    console.log("Testing backend connection...");
    
    // Test 1: Health endpoint
    const healthRes = await fetch("http://localhost:8000/health");
    console.log("Health check:", await healthRes.json());
    
    // Test 2: API docs
    const docsRes = await fetch("http://localhost:8000/docs");
    console.log("Docs accessible:", docsRes.ok);
    
    // Test 3: Try registration directly
    const testRes = await fetch("http://localhost:8000/api/v1/auth/register", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        email: "test@example.com",
        password: "Test123456",
        username: "testuser",
        full_name: "Test User"
      })
    });
    
    console.log("Registration test status:", testRes.status);
    const data = await testRes.json().catch(() => ({}));
    console.log("Registration response:", data);
    
    return { success: true, health: healthRes.ok, docs: docsRes.ok, register: testRes.ok };
  } catch (error) {
    console.error("Backend test failed:", error);
    return { success: false, error: error.message };
  }
}

File: smart_planner_web_backend/test_api.py
#!/usr/bin/env python3
import requests
import json

# Login
print("Testing login...")
login_resp = requests.post('http://localhost:8000/api/v1/auth/login', json={'email': 'testuser@example.com', 'password': 'testpass123'})
print(f'Login status: {login_resp.status_code}')
login_data = login_resp.json()
print(f'Login response: {json.dumps(login_data, indent=2)}')
token = login_data['access_token']

# Get tasks with token
print("\nTesting /tasks endpoint with token...")
headers = {'Authorization': f'Bearer {token}'}
print(f'Headers: {headers}')
tasks_resp = requests.get('http://localhost:8000/api/v1/tasks?per_page=200', headers=headers)
print(f'Tasks status: {tasks_resp.status_code}')
print(f'Tasks response: {json.dumps(tasks_resp.json(), indent=2)}')
